Server/src/redis/MsgBusImpl_redis.cpp

```diff
@@ -0,0 +1,243 @@
+#include <cstdlib>
+#include <cstring>
+#include <iostream>
+
+#include <cinttypes>
+
+#include <netdb.h>
+#include <unistd.h>
+
+#include <thread>
+#include <arpa/inet.h>
+
+#include "MsgBusImpl_redis.h"
+#include "RedisManager.h"
+
+
+
+
+
+using namespace std;
+
+/******************************************************************//**
+ * \brief This function will initialize and connect to Kafka.
+ *
+ * \details It is expected that this class will start off with a new connection.
+ *
+ *  \param [in] logPtr      Pointer to Logger instance
+ *  \param [in] cfg         Pointer to the config instance
+ ********************************************************************/
+MsgBusImpl_redis::MsgBusImpl_redis(Logger *logPtr, Config *cfg, BMPListener::ClientInfo *client) {
+    logger = logPtr;
+    this->cfg = cfg;
+    redisMgr_.Setup(logPtr, client);
```

Comment:
>Setup [](http://example.com/codeflow?start=14&length=5)

This is a big design decision to subscribe ConfigDB. Discussing offline.

Reply:
Brought-up https://github.com/sonic-net/sonic-host-services/pull/121 for monitoring config_db as separate daemon:
1> bmpcfgd will subscribe config_db table enablement, like bgp_neighbor_table/bgp_rib_in_table/bgp_rib_out_table, once config is changed, bmpcfgd will stop openbmpd, clear bmp related tables, then restart openbmpd
> openbmpd will be daemon which accept connection from bgp passively, and populate state_db for relevant bmp table like BGP_NEIGHBOR, etc.